src/hash.cc

```diff
@@ -41,7 +41,11 @@ size_t hash_data(const char* input, size_t len)
     for (ptrdiff_t i = -nblocks; i; ++i)
     {
         uint32_t key;
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
         memcpy(&key, blocks + 4*i, 4);
+#else
+        key = blocks[4*i] + (blocks[4*i + 1] << 8) + (blocks[4*i + 2] << 16) + (blocks[4*i + 3] << 24);
```

Comment:
I'd place the most significant bit first:

	const uint8_t *block = blocks + 4*i;
	uint32_t key = (block[3] << 24) + (block[2] << 16) + (block[1] << 8) + block[0];

Reply:
Yep, always using that line would be my preference, on little endian machines this is just a 32 byte load and as @Screwtapello shown it gets compiled as such.